sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getInProgressApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,409 @@
+    const studentCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    const partnerCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+        supportingUser: partner,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+
+    await db.craIncomeVerification.save([
+      studentCRAIncomeVerification,
+      partnerCRAIncomeVerification,
+    ]);
+
+    // Mock assessment service to return the saved student assessment.
+    await mockOutstandingAssessment(appModule, application.currentAssessment);
+
+    const endpoint = `/students/application/${application.id}/in-progress`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        id: application.id,
+        applicationStatus: application.applicationStatus,
+        pirStatus: application.pirStatus,
+        studentIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        partnerIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        partnerInfo: SuccessWaitingStatus.Success,
+        outstandingAssessmentStatus: SuccessWaitingStatus.Success,
+      });
+  });
+
+  it("Should get application in-progress details when there is an in-progress student assessment for an offering with later study start date.", async () => {
+    // Arrange
+
+    const currentApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+
+    // Create a previous application with an offering whose study start date and end date are much later
+    // than the current offering ones.
+    const futureApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+    const futureOffering = await db.educationProgramOffering.findOne({
+      select: {
+        id: true,
+        studyStartDate: true,
+        studyEndDate: true,
+      },
+      where: {
+        id: futureApplication.currentAssessment.offeringId,
+      },
+    });
+    futureOffering.studyStartDate = getISODateOnlyString(addDays(60));
+    futureOffering.studyEndDate = getISODateOnlyString(addDays(90));
+    await db.educationProgramOffering.save(futureOffering);
+
+    // Create an in-progress student assessment for the previous offering.
+    const futureAssessment = createFakeStudentAssessment(
+      {
+        auditUser: futureApplication.student.user,
+        application: futureApplication,
+        offering: futureOffering,
+      },
+      {
+        initialValue: {
+          calculationStartDate: addDays(70),
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+        },
+      },
+    );
+    await db.studentAssessment.save(futureAssessment);
```

Comment:
This creates a student assessment, but it does not make the newly created assessment as current assessment for the application.

Do we really need this.

And also `calculationStartDate: addDays(70)` is not required for the test.

Reply:
This test case is to check that the endpoint can successfully return the `outstandingAssessmentStatus: SuccessWaitingStatus.Success` for a the current application with the earliest offering study start date when another application (future) is created with offering start date is later than the current assessment. For this to happen, we need to mock the method `getOutstandingAssessmentsForStudentInSequence` to return the current assessment, because the database has assessments with earlier offering study start date.

Reply:
This new assessment created here does not become the application's current assessment just because the application is passed in the create factory `createFakeStudentAssessment`. The test cases are still running based on current assessment created from 
`saveFakeApplication` at line 241.